### PR TITLE
Upload/delete repo files

### PR DESCRIPTION
**Pull request by mishaschwartz**
Adds functions to call the new routes introduced in https://github.com/MarkUsProject/Markus/pull/3934

### CODE REVIEW

File: markusapi/markusapi.py [hunk 1]
```diff
@@ -238,6 +238,54 @@ def update_marking_state(self, assignment_id: int, group_id: int, new_marking_st
...
+        For example, if you want to upload a file to A1/somesubdir/myfile.txt then the short identifier
+        of the assignment with id assignment_id should be A1 and the file_path argument should be:
+        'somesubdir/myfile.txt'
+        """
+        path = Markus.get_path(assignments=assignment_id, groups=group_id, submission_files=None)
+    
+        if mime_type is None:
+            mime_type = mimetypes.guess_type(file_path)[0]
+        
+        if mime_type is None:
```
Comment (david-yz-liu) [inline markusapi/markusapi.py]: Put this block inside the above `if`.

File: markusapi/markusapi.py [hunk 2]
```diff
@@ -238,6 +238,54 @@ def update_marking_state(self, assignment_id: int, group_id: int, new_marking_st
...
+            raise ValueError(f'if the mime_type argument is not given you must provide a title file with a valid extension')
```
Comment (david-yz-liu) [inline markusapi/markusapi.py]: This is copied over from the feedback file method, and it would be good to make another commit that changes both of these in the following ways:

1. Don't need the `f` string here.
2. I'm not sure what's meant by "title file" but you can change it to "filename" or "`file_path` argument".

File: markusapi/markusapi.py [hunk 3]
```diff
@@ -238,6 +237,53 @@ def update_marking_state(self, assignment_id: int, group_id: int, new_marking_st
...
+        For example, if you want to upload a file to A1/somesubdir/myfile.txt then the short identifier
+        of the assignment with id assignment_id should be A1 and the file_path argument should be:
+        'somesubdir/myfile.txt'
+        """
+        path = Markus.get_path(assignments=assignment_id, groups=group_id, submission_files=None)
+    
+        if mime_type is None:
+            mime_type = mimetypes.guess_type(file_path)[0]        
+            if mime_type is None:
+                raise ValueError('if the mime_type argument is not given you must provide a title file_path with a valid extension')
```
Comment (david-yz-liu) [inline markusapi/markusapi.py]: I think maybe you misunderstood me. The `file_path` is good, but I want you to remove the word "title" as well. Also you should make the same change to the message on line 165.

Comment (mishaschwartz) [reply markusapi/markusapi.py]: I misunderstood.  The one on 165 should remain `'title'` though since that is the name of the variable the error message is referencing.
